crates/vfs/src/loader.rs

```diff
@@ -137,10 +137,7 @@ impl Directories {
     /// Returns `true` if `path` is included in `self`.
     pub fn contains_file(&self, path: &AbsPath) -> bool {
         let ext = path.extension().unwrap_or_default();
-        if self.extensions.iter().all(|it| it.as_str() != ext) {
-            return false;
-        }
-        self.includes_path(path)
+        self.extensions.iter().any(|it| it.as_str() == ext) && self.includes_path(path)
```

Comment:
Here, splitting a complex conditional into two parts was intentional. We *can* mechanically merge them into one boolean expression, but it's useful to show that the conditions aer logically independent. Rather than merging them, I'd add a blank like after the `}` in the original code, to signify "we've dealt with file extension, and now lets do a check for path inclusion"